### ISSUE TITLE
Feature proposal: Check problems in spec files already on mount / spec-mount

### ISSUE THREAD

**Issue by qwepoizt** [ElektraInitiative/libelektra#3987]
# Notes
* I'd like to fix this as part of my thesis.
* Is this a sensible improvement or is there a reason for the current behavior?
* I've searched through the existing issues but found no mention of this.

# Idea
* Any errors in a spec file should be detected as soon as possible (e.g. on kdb mount or kdb spec-mount).
* Right now they are either detected too late or not at all.
* Examples for errors are
    * conflicting `type` and `check/type` -> detected too late, only once `kdb set` is called.
    ```
    [myfloat]
    type = float
    # Note that here the user error is not specifying `check/type = float`. It will default to `check/type = long long` which doesn't     match the syntax of `check/range = -1.0-1.0`and therefore results in an error message once kdb set is called: `Validation Syntactic: Invalid syntax: -1.0-1.0`
    check/range = -1.0-1.0
    ```
    * specifying an illegal `type`  -> not detected at all
      ```
        [myfloat]
        type = sometypethatdoesntexist
      ```

# Motivation
* It makes sense to report errors in spec files as soon as possible and not only when the specified rules are applied.
* During my work on [Elektrifying redshift](https://github.com/jonls/redshift/issues/814) I often found myself wondering why e.g. a check isn't preventing me from `kdb set`ing illegal values, until I finally found I had a typo or forgot something (see examples above)

**Comment by qwepoizt** [comment]
@markus2330 What do you think about this?

**Comment by qwepoizt** [comment]
Thanks @markus2330 for your input. I will get back to you after my work on other issues.

**Comment by qwepoizt** [comment]
Decision from call @markus2330 and @qwepoizt  03rd Sep 2021:

* This is difficult, because it requires static checks (in contrast to dynamic, during runtime). 
* An elaborate exceeds the scope of a bachelor thesis.
* Easier workaround fixes are possible, e.g. implementing some checks in the spec-mounter.
* @qwepoizt will look into easier fixes.

**Comment by qwepoizt** [comment]
@markus2330 As discussed, I've implemented several sanity checks in the spec-mounter with #4057 !

I think this is a good solution for this issue and #3983: When someone creates a specification file and then executes `kdb spec-mount ...` to test/use it, they will receive errors if one of the sanity checks fails!

I think I can say from experience that these checks will help a lot, especially if someone is just getting started with Elektra/specification files. They cover the mistakes I made while writing the specification for redshift.
